### PR TITLE
#14 - check package versions

### PR DESCRIPTION
**Pull request by mengjiez99**
https://github.com/mgibbs1259/gigantum_testing/issues/14

Check the versions of packages from the environment, then implement codes in jupyter notebook to print out versions and assure they match.

Motivation:
It would be good to implement codes in jupyter notebook in gigantum to print out versions of packages installed. and assure that they match the version in the environment.

### CODE REVIEW

**Comment by billvb** [comment]
One general comment -- it will be very good for your future careers to get familiar with PEP-8 https://www.python.org/dev/peps/pep-0008/

Not today at work, but in your free time please read it over carefully. It is _the_ definitive python coding style guide -- detailing how to do comments, how long lines should be, when and how to split lines. Going forward, we need to be more careful to adhere to this standard. There are actually tools that you can link to Github that will reject any commits that are not PEP-8 compliant. We're not going to to that for this repository yet, but I think it'll be good to code as if we would

File: create_project.py [hunk 1]
```diff
@@ -280,20 +283,36 @@ def test_pip_packages(driver):
     wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, ".flex>.Stopped")))
     assert driver.find_element_by_css_selector(".flex>.Stopped").is_displayed(), "Expected stopped container"
 
+    # check package version from environment
```
Comment (billvb) [inline create_project.py]: Remove prints and add more logging calls to indicate what its doing, and what it got

File: create_project.py [hunk 2]
```diff
@@ -280,20 +283,36 @@ def test_pip_packages(driver):
...
+    package_info = driver.find_element_by_css_selector(".PackageDependencies__table-container").text.split("\n")[1::2]
```
Comment (billvb) [inline create_project.py]: This line needs to be broken up, and needs a comment why it's doing that [1::2] at the end

File: create_project.py [hunk 3]
```diff
@@ -280,20 +283,36 @@ def test_pip_packages(driver):
...
     print("switch success")
-    time.sleep(10)
+    time.sleep(5)
     driver.find_element_by_css_selector("[data-category = Notebook]").click()
-    time.sleep(10)
-    driver.find_element_by_css_selector(".CodeMirror-line").click().send_keys("import pandas")
+    time.sleep(5)
+    el = driver.find_element_by_css_selector(".CodeMirror-line")
+    actions = ActionChains(driver)
+    actions.move_to_element(el).click(el).send_keys("import pandas, numpy, matplotlib\nprint('pandas', pandas.__version__,\n'numpy',numpy.__version__,\n'matplotlib', matplotlib.__version__)").perform()
```
Comment (billvb) [inline create_project.py]: This line needs to be broken up

File: create_project.py [hunk 4]
```diff
@@ -280,20 +283,36 @@ def test_pip_packages(driver):
...
+    time.sleep(5)
+    driver.find_element_by_css_selector(".jp-RunIcon").click()
+    output = driver.find_element_by_css_selector(".jp-OutputArea-output").text.split(" ")
```
Comment (billvb) [inline create_project.py]: This whole bit should be re-formatted too with more descriptive variable names and something a little less cryptic than `dict(zip(output[::2], output[1::2]))`

**Comment by billvb** [comment]
Also please add .DS_Store to .gitignore if it's not already there

File: create_project.py [hunk 5]
```diff
@@ -280,20 +283,42 @@ def test_pip_packages(driver):
...
+    # convert to dictionary with package names as key and versions as values.
+    package_jupyter = dict(zip(package_output[::2], package_output[1::2]))
+    logging.info("Getting package versions from jupyterlab")
+    # check if package versions from environment and from jupyter notebook are same.
+    assert package_environment == package_jupyter, "Package versions match"
     time.sleep(10)
-    driver.find_element_by_css_selector(".CodeMirror-line").click().send_keys("import pandas")
-    time.sleep(20)
 
+    '''
```
Comment (billvb) [inline create_project.py]: Let's get rid of this commented-out code if you don't need it any more

Comment (mengjiez99) [reply create_project.py]: I think we need to work on that on later when the container status bug is resolved for installing conda packages.

**Pull request merged**